xcvm/architecture/transports/otp.md

```diff
@@ -0,0 +1,208 @@
+
+1. Fungible token transfers.
+2. Opaque data transfers.
+
+Ideally, a protocol supports both in a single operation, although most bridges will need to be wrapped in an additional set of contracts to be fully compatible.
+
+`OTP` handles the following operations:
+
+1. Maintaining a blacklist of transports.
+2. Ingress of messages on the chain.
+3. Registration of routing.
+4. Egress of messages on the chain. 
+
+## Blacklist
+
+The blacklist is used to enact emergency operations in case of bridge fraud, which can be caused by hacks, or bugs within the system. The blacklist globally disables bridges based on `LAYR` governance.
+
+Note that the blacklist is not used to manage ecosystem specific risk, such as toxic assets. It only concerns itself with bridges where the message authenticity can no longer be guaranteed.
+
+## XCVM Ingress
+
+A transport does not need to register itself to pass messages into the `XCVM`, it can do so directly, assuming that the interpreter instance accepting the transport's messages has whitelisted this transport. 
+
+The following diagram describes the flow of data (and funds) when OTP is used to for ingress of programs.
+
+```mermaid
+sequenceDiagram
+    Third-party transport->>Third-party OTP-adapter: Transform messsage.
+    Third-party OTP-adapter->>Gateway: Transfer assets and pass program.
+    Gateway->>Router: Add bridging info and transfer funds.
+    Note over Gateway: Accept/deny program based on blacklist.
+    Router->>XCVM Interpreter: Instantiate vm and transfer funds.
+    Note over XCVM Interpreter: Accept/deny program based on gateway annotation.
+```
+
+Although the diagram simplifies the work that the OTP-adapter might need to perform, it can be relatively complicated depending on how the transport operates. We go into more concrete implementations [later](#possible-implementations) in this document. 
+
+## Registration of Routes
+
+For a transport to be possible be selected by the Gateway to transfer messages to different chains, a more involved registration process is required. 
+
+- A stake needs to be put down, which is used to reward disputers in case of bridge fraud.
+- Metadata on the bridge must be provided, including 
+    1. The `Account` of the administrator, used to deregister the bridge, and add routes.
+    2. Destinations reachable through this bridge.
+
+```
+RegistrationInfo ::= Account [Destination]
+Destination      ::= NetworkId [AssetId]
+```
+
+Immediately after registration, the bridge registration can be disputed for approximately 72 hours. The stake can be lost during this time, if governance determines that the registration is fraudulent. If a dispute is triggered, the registration can be delayed by up to 2 weeks.
+
+#### Disputed Registration
+
+The following diagram displays the sequence of a registration, including the dispute by a validator.
+
+```mermaid
+sequenceDiagram
+    Third-party Transport->>Gateway: Register bridge for egress.
```

Comment:
is Gateway will be only one chain as some root (Composable/Picasso), and stuff copied to other chains? I am thinkin about if XCVM will be initiated on other chains?

Reply:
No, we regularly deploy the contracts; and set the admin to Identity: (NetworkId::Composable, SomeContractAddress), which can then use IBC to configure the gateway further. `SomeContractAddress` can actually be our democracy pallet for example